Data/ByteString.hs

```diff
@@ -442,17 +442,18 @@ append = mappend
 -- | /O(n)/ 'map' @f xs@ is the ByteString obtained by applying @f@ to each
 -- element of @xs@.
 map :: (Word8 -> Word8) -> ByteString -> ByteString
-map f (BS fp len) = unsafeDupablePerformIO $ unsafeWithForeignPtr fp $ \a ->
-    create len $ map_ 0 a
+map f (BS fp len) = unsafeDupablePerformIO $ unsafeWithForeignPtr fp $ \ptr1 ->
+    create len $ \ptr2 -> m ptr1 ptr2
```

Comment:
Let's take a chance to use better names. I suggest `srcPtr` and `dstPtr` or something similar.

Reply:
Will do!